model/task_lifecycle.go

```diff
@@ -1451,28 +1451,23 @@ func ClearAndResetStrandedTask(h *host.Host) error {
 		return errors.WithStack(MarkEnd(t, evergreen.MonitorPackage, time.Now(), &t.Details, false))
 	}
 
+	// Only restart failed tasks for stranded tasks.
+	t.SetRestartFailed()
```

Comment:
should only restart stranded but this will restart other tasks that have failed 
I think this behavior is preferable to the current one and can be amended in the future ticket to make stranded tasks work but I can remove this if need be